tools/testgrid-analysis/cmd/data.go

```diff
@@ -150,6 +202,24 @@ func intStatusSet(statuses []statuspb.TestStatus) map[int32]struct{} {
 }
 
 func shortenTestName(fullname string) string {
-	parts := strings.Split(fullname, ".")
-	return parts[len(parts)-1]
+	parts := strings.Split(fullname, "/")
+	// keep the package name of the test.
+	if len(parts) < 2 || strings.Contains(parts[len(parts)-1], ".") {
+		return parts[len(parts)-1]
+	}
+	return parts[len(parts)-2]
+}
+
+func mergeSet(from, to map[string]struct{}) {
```

Comment:
Maybe `mergeMaps(...)` is a better name for the function, as technically, they are maps in Go.